src/solvers/refinement/string_refinement.cpp

```diff
@@ -26,10 +26,34 @@ Author: Alberto Griggio, alberto.griggio@gmail.com
 #include <solvers/sat/satcheck.h>
 #include <solvers/refinement/string_constraint_instantiation.h>
 #include <java_bytecode/java_types.h>
+#include <util/optional.h>
+
+#include <util/ssa_expr.h> // for char array pointer
```

Comment:
Both of these appear to be unused.

Reply:
Not sure, there clearly is a usage of optional in the file for example. Question is whether we should start removing headers if they were already included in parent file. Or make some sort of `stdafx.h` precompiled header file, which would include most commonly used headers (STL utilities, containers, strings, etc.)

Reply:
I went to go look in [`CODING_STANDARD.md`](https://github.com/diffblue/cbmc/blob/develop/CODING_STANDARD.md#header-files). It's not completely clear on this type of case, but it does say avoid unnecessary `#includes`, which they are in the literal sense.